homeassistant/components/calendar/__init__.py

```diff
@@ -200,12 +201,12 @@ async def get(self, request, entity_id):
         start = request.query.get("start")
         end = request.query.get("end")
         if None in (start, end, entity):
-            return web.Response(status=HTTP_BAD_REQUEST)
+            return web.Response(status=HTTPStatus.BAD_REQUEST.value)
```

Comment:
No need to access the value.

```python
from http import HTTPStatus
from aiohttp.web import Response

resp = Response(status=HTTPStatus.BAD_REQUEST)
resp.status
400
```

Reply:
I know, but to me that's just a lucky coincidence, status is type hinted to be an `int`: https://github.com/aio-libs/aiohttp/blob/d70655445e0ceaa90c548ecc826b6873335f622e/aiohttp/client_reqrep.py#L652

There's no guarantee that passing an `IntEnum` to something that takes an `int` will work as if an `int` was passed.
```
>>> from http import HTTPStatus
>>> def foo(status):
...  print(status)
... 
>>> foo(HTTPStatus.OK)
HTTPStatus.OK
```

Reply:
It's always converted to int.

Reply:
https://docs.python.org/3/library/enum.html#enum.IntEnum

>     Base class for creating enumerated constants that are also subclasses of int.

Reply:
I can only say that my example clearly proves that it is _not_ always equivalent to or converted to an int. If it was, the output would be `200`, but it's `HTTPStatus.OK`. More examples:
```
>>> from http import HTTPStatus
>>> HTTPStatus.OK
<HTTPStatus.OK: 200>
>>> str(HTTPStatus.OK)
'HTTPStatus.OK'
>>> "%s" % HTTPStatus.OK
'HTTPStatus.OK'
```

I don't think there is anything wrong with doing a `str()` on it, but it doesn't behave like a regular `int` when one does it. In that sense it doesn't matter if it's an `int` subclass or not, one can not just pass it anywhere where an `int` is taken, chances are that it'll break stuff.

But as we've observed, in the aiohttp case it appears to not break right now. If we insist on taking the risk, I can modify the PR to drop the `.value`, but would not do it if it was my call, without an additional confirmation from aiohttp folks that doing so is ok and won't break. (They don't seem terribly active so I don't feel like asking and waiting.)

Reply:
Looking at the source for aiohttp:
1. The `status: int` argument to the constructor of the `Response` class is only used in being sent to the parent class https://github.com/aio-libs/aiohttp/blob/42881feb6c9b5bb7c74f1b3fa05b5bdec49120e6/aiohttp/web_response.py#L571
2. The constructor for the parent class (`StreamResponse`) only uses the `status` argument in a call to `self.set_status(status, reason)` https://github.com/aio-libs/aiohttp/blob/42881feb6c9b5bb7c74f1b3fa05b5bdec49120e6/aiohttp/web_response.py#L124
3. In `set_status`, the `status` argument is cast to an `int` before being used: https://github.com/aio-libs/aiohttp/blob/42881feb6c9b5bb7c74f1b3fa05b5bdec49120e6/aiohttp/web_response.py#L162
4. By my testing, casting a member of `IntEnum` to an `int` (e.g. `int(MyIntEnum.Foo)`) will, in fact, result in an `int`.

Reply:
As said, I can modify this PR to drop the `.value` if that's what it takes to get this in. Just waiting for the confirmation before starting to make changes.

Reply:
Let's drop the value.

Reply:
Done + rebased to resolve conflicts.